californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java

```diff
@@ -196,10 +202,15 @@ public void receiveRequest(final Exchange exchange, final Request request) {
 
 		if (request.isDuplicate()) {
 			// Request is a duplicate, so resend ACK, RST or response
-			if (exchange.getCurrentResponse() != null) {
+			Response currentResponse = exchange.getCurrentResponse();
+			if (currentResponse != null) {
 				LOGGER.debug("respond with the current response to the duplicate request");
+				if (currentResponse.isConfirmable()) {
+					// Simulate retransmission cycle, CON responses requires this
+					exchange.setFailedTransmissionCount(1);
+				}
```

Comment:
The [RFC-7252](https://tools.ietf.org/html/rfc7252#section-4.2) say :

_"A CoAP endpoint that sent a Confirmable message MAY give up in
   attempting to obtain an ACK even before the MAX_RETRANSMIT counter
   value is reached.  For example, the application has canceled the
   request as it no longer needs a response, or there is some other
   indication that the CON message did arrive.  In particular, a CoAP
   request message may have elicited a separate response, in which case
   it is clear to the requester that only the ACK was lost and a
   retransmission of the request would serve no purpose.  **However, a
   responder MUST NOT in turn rely on this cross-layer behavior from a
   requester**, i.e., it MUST retain the state to create the ACK for the
   request, if needed, even if a Confirmable response was already
   acknowledged by the requester."_

So 2 questions : 
1. For separated response, does it means that we should send a ACK before ?
2.  Should we even consider to just send the ACK and rely on CON retransmission for the response ?

Reply:
During tests a exception occurs and therefore I just suppressed that. 
To comply to that, some more work must be done :-).